### PR TITLE
Use strlen to truncate CSTRING values

### PR DESCRIPTION
**Pull request by nirbhayc**
https://github.com/bloomberg/comdb2/issues/1038

### CODE REVIEW

File: db/sqlinterfaces.c [hunk 1]
```diff
@@ -2711,11 +2698,18 @@ static int bind_parameters(struct reqlogger *logger, sqlite3_stmt *stmt,
             break;
         case CLIENT_CSTR:
         case CLIENT_PSTR:
-        case CLIENT_PSTR2:
-            fix_cstr_len((char *)p.u.p, &p.len);
+        case CLIENT_PSTR2: {
+            /* Keeping with the R6 behavior, we need to 'silently' truncate
+             * the string on first '\0'.
+             */
+            size_t len = strlen((char *)p.u.p);
```
Comment (akshatsikarwar) [inline db/sqlinterfaces.c]: Input may not be null terminated. `p.len = strnlen(p.u.p, p.len)`?

Comment (nirbhayc) [reply db/sqlinterfaces.c]: good point

File: db/sqlinterfaces.c [hunk 2]
```diff
@@ -2702,10 +2702,7 @@ static int bind_parameters(struct reqlogger *logger, sqlite3_stmt *stmt,
             /* Keeping with the R6 behavior, we need to 'silently' truncate
              * the string on first '\0'.
              */
-            size_t len = strlen((char *)p.u.p);
-            if (len < p.len) {
-                p.len = len;
-            }
+            size_t len = strnlen((char *)p.u.p, p.len);
```
Comment (akshatsikarwar) [inline db/sqlinterfaces.c]: Assign to `p.len`, no?

Comment (nirbhayc) [reply db/sqlinterfaces.c]: WHOA

Comment (akshatsikarwar) [reply db/sqlinterfaces.c]: `0xf00d 0xc0ffee`

Comment (nirbhayc) [reply db/sqlinterfaces.c]: Or probably cuz Monday.. :s

**Comment by akshatsikarwar** [comment]
LGTM, but please check why cbuild failed.

**Comment by nirbhayc** [comment]
@akshatsikarwar Cbuild seem to be failing randomly. Thanks for the review!

**Pull request merged**